force-app/main/default/classes/TestRunProcessor.cls

```diff
@@ -103,11 +103,40 @@ public with sharing class TestRunProcessor implements Schedulable {
                     Method_Pass__c = apexTestResult.Outcome == 'Pass' ? true : false,
                     Name = apexTestResult.ApexClass.Name,
                     Stack_Trace__c = apexTestResult.stackTrace,
-                    Test_Run__c = testRun.Id
+                    Test_Run__c = testRun.Id 
                 )
             );
+            
         }
-
-        insert results;
+        
+		List<Test_Run_Method_Result__c> result = processPassFailDates(results);
+        insert result;
+    }
+    
+    private List<Test_Run_Method_Result__c> processPassFailDates(List<Test_Run_Method_Result__c> result) {
+        List<String> failedTests = new List<String>();
+        for (Test_Run_Method_Result__c r : result) {
+            if (!r.Method_Pass__c) {
+                failedTests.add(r.Method_Name__c);
+            }
+        }      
+        List<Test_Run_Method_Result__c> failedCousins = [SELECT Id, Method_Name__c, Method_Pass__c, CreatedDate FROM Test_Run_Method_Result__c 
+                                                   WHERE Method_Name__c IN :failedTests ORDER BY CreatedDate DESC];
+        
+        for (Test_Run_Method_Result__c r : result) {
```

Comment:
shouldn't we just be iterating over `failedTests` here?

Reply:
No because I am passing back the entire set of tests including those that did not fail.

Reply:
Why?

This method doesn't need to return anything, just modify the elements have failed and need the additional information.

Reply:
So after I modify should I insert again? As of now I am calling this method in "processTestResults" and inserting the returned sObjects

Reply:
you don't need to run DML again.  The basic steps are:

- create the records
- find any failures
- determine lastSuccess/firstFailure
- set fields
- insert records

Your code is doing this, but just looping over records and making additional assignments needlessly